.github/workflows/build.yml

```diff
@@ -0,0 +1,195 @@
+name: CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+# cancel already running builds of the same branch or pull request
+concurrency:
+  group: ci-${{ github.head_ref }} || concat(${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CONFIGURE_FLAGS: --enable-compile-warnings=maximum
+  CFLAGS: -g -O2 -Werror=pointer-arith -Werror=implicit-function-declaration
+  JOBS: 2
+  DEBUG: 1
+  # Useful URL: https://github.com/mate-desktop/debian-packages
+  # Useful URL: https://salsa.debian.org/debian-mate-team/mate-panel
+  DEB_LIBRARY_DEPS: |
+    libatk1.0-dev
+    libcairo2-dev
+    libdconf-dev
+    libgirepository1.0-dev
+    libglib2.0-dev
+    libgtk-3-dev
+    libgtk-layer-shell-dev
+    libice-dev
+    libmate-desktop-dev
+    libmate-menu-dev
+    libmateweather-dev
+    libpango1.0-dev
+    libsm-dev
+    libsoup2.4-dev
+    libwnck-3-dev
+    libx11-dev
+    libxrandr-dev
+    libgirepository1.0-dev
+  DEB_BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    autopoint
+    gir1.2-freedesktop
+    git
+    gobject-introspection
+    gtk-doc-tools
+    lsb-release
+    make
+    mate-common
+    meson
+    yelp-tools
+    iso-codes
+    gobject-introspection
+  # TODO
+  DEB_SCAN_BUILD_DEPS: |
+    clang
+    clang-tools
+  # Useful URL: https://git.archlinux.org/svntogit/community.git/tree/mate-panel
+  # + mate-desktop dependencies
+  ARCH_BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    clang
+    gcc
+    git
+    gobject-introspection
+    gtk-layer-shell
+    itstool
+    libcanberra
+    libmateweather
+    libsm
+    libwnck3
+    make
+    mate-common
+    mate-desktop
+    mate-menus
+    meson
+    which
+    yelp-tools
+    iso-codes
+    gobject-introspection
+
+jobs:
+  build:
+    name: Build on ${{matrix.container}} with in-process=${{matrix.in-process}}
+    runs-on: ubuntu-latest
+    container: ${{matrix.container}}
+
+    strategy:
+      fail-fast: false
```

Comment:
This line means that we'll keep other builds in the matrix running even if one fails.  Is it needed, though?  Upside is that we can see if it's distro-specific, downside is that if it's a plain code mistake it'll run other ones for nothing…

Reply:
We should catch distro-specific issues, ihmo.

Reply:
In either case we will, the only question is whether one build failure cancels the other ones, or whether they all finish.

Reply:
I meant when eg. the job debian-testing in-process failed all other jobs should finished if possible.
Otherwise i have to look in all job logs to see which job failed. Or do i see something like canceled because a other job failed?

Reply:
I guess you'd see other jobs were cancelled (and possibly why?), but I'm not entirely sure.

Reply:
I would prefer to let other job finished. If other will be stopped because of an error of one job we don't know that all other jobs are free from errors.

Reply:
Yeah, that was my thinking, so we can leave it as is